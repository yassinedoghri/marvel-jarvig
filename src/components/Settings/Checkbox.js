import styled from "styled-components";

const Label = styled.label`
  display: inline-block;
  margin-top: 2em;
  width: 100%;

  & input {
    margin-right: 0.5em;
  }
`;

export default Label;
