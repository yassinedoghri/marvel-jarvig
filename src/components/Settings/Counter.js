import styled from "styled-components";

const Label = styled.p`
  display: inline-block;
  float: right;
`;

export default Label;
