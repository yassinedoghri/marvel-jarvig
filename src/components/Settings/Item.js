import styled from "styled-components";

const Item = styled.div`
  position: relative;
`;

export default Item;
