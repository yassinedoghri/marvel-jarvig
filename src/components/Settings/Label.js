import styled from "styled-components";

const Label = styled.p`
  display: inline-block;
  margin-bottom: ${props => props.theme.spaces.sm}em;
`;

export default Label;
