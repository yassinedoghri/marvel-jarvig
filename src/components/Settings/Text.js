import styled from "styled-components";

const Text = styled.p``;

export default Text;
