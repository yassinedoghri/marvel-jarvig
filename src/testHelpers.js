import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
