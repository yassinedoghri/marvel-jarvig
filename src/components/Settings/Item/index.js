import styled from 'styled-components';

import Label from './Label';
import Counter from './Counter';

const Item = styled.div`
    position: relative;
`;

Item.Label = Label;
Item.Counter = Counter;

export default Item;
