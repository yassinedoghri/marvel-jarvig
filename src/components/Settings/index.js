import Checkbox from "components/Settings/Checkbox";
import Counter from "components/Settings/Counter";
import Item from "components/Settings/Item";
import Label from "components/Settings/Label";

import styled from "styled-components";

const Settings = styled.div``;

Settings.Item = Item;
Settings.Label = Label;
Settings.Counter = Counter;
Settings.Checkbox = Checkbox;

export default Settings;
