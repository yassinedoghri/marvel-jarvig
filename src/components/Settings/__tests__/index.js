import Settings from "components/Settings";
import React from "react";
import { renderWithTheme } from "utils/testHelpers";

describe("Settings Block", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <Settings>
        <Settings.Item>
          <Settings.Label>Game time (min)</Settings.Label>
          <Settings.Counter />
        </Settings.Item>
        <Settings.Item>
          <Settings.Checkbox>
            <input type="checkbox" />Checkbox
          </Settings.Checkbox>
        </Settings.Item>
      </Settings>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
