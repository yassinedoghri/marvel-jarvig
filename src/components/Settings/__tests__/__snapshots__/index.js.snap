// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Block renders correctly 1`] = `
<div
  className="sc-ifAKCX cFlEyZ"
>
  <div
    className="sc-htpNat fjcJqy"
  >
    <p
      className="sc-bxivhb juXbqX"
    >
      Game time (min)
    </p>
    <p
      className="sc-bwzfXH iHMWje"
    />
  </div>
  <div
    className="sc-htpNat fjcJqy"
  >
    <label
      className="sc-bdVaJa jTaxDY"
    >
      <input
        type="checkbox"
      />
      Checkbox
    </label>
  </div>
</div>
`;
